File: CivitasRuby/pkg/CivitasRuby-0.0.1/lib/casilla_sorpresa.rb
#encoding:utf-8

require_relative 'casilla'
require_relative 'sorpresa'
require_relative 'mazo_sorpresas'
require 'casilla.rb'

module Civitas
    class CasillaSorpresa < Casilla
        def initialize(mazo, nombre)
            super(nombre)
            @mazo=mazo
        end

        def recibe_jugador(actual, todos)
            if(jugador_correcto(actual,todos))
                todos[actual].puede_comprar=false
                informe(actual,todos)
                @sorpresa=@mazo.siguiente
                @sorpresa.aplicar_a_jugador(actual, todos)
            end
        end

        def to_string
                String::new(str="**************************************************")
                str+="\n " + @nombre + " ¯|(º o)/¯"
                str+="\n**************************************************\n"

                return str;
        end
    end
end

File: CivitasJava/src/civitas/CasillaSorpresa.java
package civitas;

import java.util.ArrayList;

public class CasillaSorpresa extends Casilla{

    CasillaSorpresa(MazoSorpresas mazo, String nombre){
        super(nombre);
        this.mazo=mazo;
    }

    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            todos.get(actual).puedeComprar=false;
            informe(actual, todos);
            sorpresa=mazo.siguiente();
            sorpresa.aplicarAJugador(actual, todos);
        }
    }

    @Override
     public String toString(){

        String str="******************************";
        str+="\n " + getNombre() + " ¯|(º o)/¯";
        str+="\n******************************\n";
    
        return str;
    }
}